index.ts

```diff
@@ -732,6 +786,190 @@ export const ReleaseCollection = {
+      console.log("Unknown Endpoint:", path);
+  }
+}
+async function register(owner: string, repo: string, event: string) {
+  const webhookURL = (await nodes.endpoint.$get()) + "/webhooks";
+  try {
+    // Check if the repository already has a webhook
+    const repository = state.repos[`${owner}/${repo}`];
+    const { data: hooks } = await client().repos.listWebhooks({
+      owner,
+      repo,
+    });
+    const matchingHook = hooks.find((hook) => hook.id === repository.id);
+    // If the repository already has a webhook, update it
+    if (matchingHook) {
+      if (repository.events.includes(event)) {
+        console.log("Webhook already exists with the same event.");
+        return;
+      } else {
+        const updatedEvents = [...repository.events, event];
+        await client().repos.updateWebhook({
+          owner,
+          repo,
+          hook_id: repository.id,
+          config: {
+            content_type: "json",
+            url: webhookURL,
+          },
+          events: updatedEvents,
+        });
+        // Update the events array in the repository object
+        repository.events = updatedEvents;
+        console.log("Webhook updated with new event.");
+        return;
+      }
+    }
+    // Create a new webhook
+    const {
+      data: { id: webhookId },
+    } = await client().repos.createWebhook({
+      owner,
+      repo,
+      events: [event],
+      config: {
+        content_type: "json",
+        url: webhookURL,
+      },
+    });
+
+    // Add the repository to the state with the webhook data
+    state.repos[`${owner}/${repo}`] = {
+      id: webhookId,
+      url: webhookURL,
+      events: [event],
+    };
+
+    console.log("New webhook created.");
+  } catch (error) {
+    throw new Error(error);
+  }
+}
+
+async function unregister(owner: string, repo: string, event: string) {
+  try {
+    // Check if the repository has a webhook
+    const repository = state.repos[`${owner}/${repo}`];
+    if (!repository) {
+      console.log(`Webhook does not exist for ${owner}/${repo}.`);
+      return;
+    }
+    // Update the webhook to remove the specified events
+    const updatedEvents = repository.events.filter((e: string) => e !== event);
+
+    await client().repos.updateWebhook({
+      owner,
+      repo,
+      hook_id: repository.id,
+      config: {
+        content_type: "json",
+        url: repository.url,
+      },
+      events: updatedEvents,
+    });
+    repository.events = updatedEvents;
+    console.log(`Event ${event} deleted from webhook.`);
+    // Delete the webhook if there are no more events
+    if (updatedEvents.length === 0) {
+      await client().repos.deleteWebhook({
+        owner,
+        repo,
+        hook_id: repository.id,
+      });
+      delete state.repos[`${owner}/${repo}`];
+      console.log("Webhook deleted.");
+      return;
+    }
+  } catch (e) {
+    throw new Error(`Failed to delete event ${event} from webhook for ${owner}/${repo}`);
```

Comment:
Probably good to include the original error as `cause` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error